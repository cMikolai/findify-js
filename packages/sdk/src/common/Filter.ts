/**
 * Filter that can be use to
 * narrow down the search query.
 */
export interface Filter {
  /** Name of filter */
  name: string;
  /** Type of filter */
  type: string;
  /** Array of selected filter values */
  values?: FilterValue[];
}

/*
 * Represents selected filter value.
 */
export interface FilterValue {
  /** Selected filter value */
  value?: string;
  /** Selected range filter "from" value */
  from?: string;
  /** Selected range filter "to" value */
  to?: string;
}
