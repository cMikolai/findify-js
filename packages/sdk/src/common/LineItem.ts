/**
 * Product line item.
 */
export interface LineItem {
  /** Item ID bought */
  item_id: string;
  /** Sale price of the product */
  unit_price: number;
  /** Quantity bought */
  quantity: number;
}
