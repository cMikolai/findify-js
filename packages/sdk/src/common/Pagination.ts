export interface Pagination {
  /**  Offset of products items. Used in pagination */
  offset?: number;
  /** Number of products in result */
  limit?: number;
}
