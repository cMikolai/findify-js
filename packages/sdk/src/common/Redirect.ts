export interface Redirect {
  name: string;
  url: string;
}
