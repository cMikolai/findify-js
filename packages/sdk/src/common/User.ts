/**
 * Defines the current user,
 * which gives just enough information to enable personalization.
 * User object has two properties that are required: `uid` and `sid`.
 */
export interface User {
  /**
   * This is a unique identifier of the user across multiple sessions.
   * You can store the uid in the cookie or get it from any other source (like a database).
   * It must uniquly identify the user.
   * */
  uid: string;
  /**
   * A unique session identifier,
   * the best is to store the value in a cookie on the client side,
   * with a TTL of 30 minutes or use a session ID that your platform provides.
   */
  sid: string;
  /**
   * The email from the user, if the user is logged in.
   * Providing an email allows us to track the user across
   * devices and give them relevant personalized results.
   */
  email?: string;
  /**
   * Not required in browser environment.
   * By default we get the ip from the request that we receive.
   * If call is made not from the user's browser,
   * please include the relevant user ip in the request.
   */
  ip?: string;
  /**
   * Not required in browser environment.
   * By default we get the user agent from the request that we receive.
   * If call is made not from the user's browser,
   * please include the relevant user user agent in the request.
   */
  ua?: string;
  /**
   * Not required in browser environment.
   * By default we get the user languages from the request that we receive.
   * If call is made not from the user's browser,
   * please include the relevant user user languages in the request.
   *
   * You can use either an ISO 639-1 language format code or
   * include the locale code: en-GB.
   */
  lang?: string[];
}
