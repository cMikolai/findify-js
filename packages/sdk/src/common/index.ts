export { Filter } from './Filter';
export { Redirect } from './Redirect';
export { Sort, SortingOrder } from './Sort';
export { User } from './User';
export { Pagination } from './Pagination';
export { LineItem } from './LineItem';
