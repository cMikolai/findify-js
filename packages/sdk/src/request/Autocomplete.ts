import { Type as RequestType } from './Type';
import { Base } from './params';

/**
 * Autocomplete request.
 */
export interface Request {
  type: RequestType.Autocomplete;
  /** Request parameters */
  params: Params;
}

/**
 * Autocomplete request parameters.
 */
export interface Params extends Base {
  /** Autocomplete query */
  q?: string;
  /** Limit of search suggestions */
  suggestion_limit?: number;
  /** Limit of product matches */
  item_limit?: number;
}
