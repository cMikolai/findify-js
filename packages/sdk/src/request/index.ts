import * as Autocomplete from './Autocomplete';
import * as Search from './Search';
import * as SmartCollection from './SmartCollection';
import * as Recommendations from './Recommendations';
import * as Feedback from './Feedback';
import * as Params from './params';

/**
 * Request method.
 */
export enum Method {
  POST = 'post',
  JSONP = 'jsonp',
}

/**
 * Represents a request.
 */
export type Request =
  | Autocomplete.Request
  | Search.Request
  | SmartCollection.Request
  | Recommendations.Request
  | Feedback.Request;

/**
 * Represents a request body.
 */
export type Body =
  | Autocomplete.Params
  | Search.Params
  | SmartCollection.Params
  | Recommendations.Params
  | Feedback.Params;

export { Type } from './Type';
export {
  Params,
  Autocomplete,
  Search,
  SmartCollection,
  Recommendations,
  Feedback,
};
