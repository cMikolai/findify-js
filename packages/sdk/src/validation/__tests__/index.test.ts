import { SortingOrder } from '../../common';
import * as R from '../../request';
import * as V from '..';

describe('validation', () => {
  describe('validateConfig', () => {
    const validate = (cfg: any, browser: boolean = true) => () =>
      V.validateConfig(cfg, browser);

    it('throws an error if "key" is missing or invalid', () => {
      expect(validate({} as any)).toThrowErrorMatchingSnapshot();
      expect(validate({ key: 123 } as any)).toThrowErrorMatchingSnapshot();
      expect(validate({ key: true } as any)).toThrowErrorMatchingSnapshot();
      expect(validate({ key: '' })).toThrowErrorMatchingSnapshot();
      expect(validate({ key: '   ' })).toThrowErrorMatchingSnapshot();
    });

    it('throws an error if "method" is invalid', () => {
      expect(
        validate({ key: 'ab3213', method: 'unsupported' })
      ).toThrowErrorMatchingSnapshot();
    });

    it('throws an error if "log" is invalid', () => {
      [null, 321, 'string'].forEach(log =>
        expect(validate({ key: 'somekey', log })).toThrowErrorMatchingSnapshot()
      );
    });

    it('throws and error if trying to use jsonp when not in browser', () => {
      expect(
        validate({ key: 'key', method: R.Method.JSONP }, false)
      ).toThrowErrorMatchingSnapshot();
    });

    it('does not throw an error if config is correct', () => {
      expect(validate({ key: 'key1' })).not.toThrow();
      expect(validate({ key: 'key2', method: R.Method.JSONP })).not.toThrow();
      expect(validate({ key: 'good', method: R.Method.POST })).not.toThrow();
      expect(validate({ key: 'smth', log: false })).not.toThrow();
      expect(validate({ key: 'othr', log: true })).not.toThrow();
    });
  });

  describe('validateAutocomlete', () => {
    it('throws error if "q" is missing or invalid', () => {
      const allowedQueries = ['bar', 32, false, true];
      testValidatePresence(V.validateAutocomplete, 'q', allowedQueries);
    });
  });

  describe('validateSearch', () => {
    it('throws error if request parameters is invalid', () => {
      const allowedQueries = ['foo', 123, true, false];
      testValidatePresence(V.validateSearch, 'q', allowedQueries);
      testValidateList(V.validateSearch, { q: 'asics' });
    });
  });

  describe('validateSmartCollection', () => {
    it('throws error if request parameters is invalid', () => {
      const allowedSlots = ['bar', 456, false, true];
      testValidatePresence(V.validateSmartCollection, 'slot', allowedSlots);
      testValidateList(V.validateSmartCollection, { slot: 'slot-handle' });
    });
  });

  describe('validateRecommendations', () => {
    it('throws error if recommendations type is missing', () => {
      const invalidTypes = [{}, { type: null }, { type: '' }, { type: '  ' }];
      invalidTypes.forEach(req =>
        expect(() =>
          V.validateRecommendations(req as any)
        ).toThrowErrorMatchingSnapshot()
      );
    });

    it('throws error if recommendation type is invalid', () => {
      expect(() =>
        V.validateRecommendations({ type: 'unsupported' } as any)
      ).toThrowErrorMatchingSnapshot();
    });

    describe('slot', () => {
      it('throws error if "slot" is missing or invalid', () => {
        const allowedSlots = ['bar', 'baz-qux', 456];
        testValidatePresence(V.validateRecommendations, 'slot', allowedSlots, {
          type: R.Recommendations.Type.Slot,
        });
      });
    });

    describe('also viewed / bought', () => {
      it('throws error if "item_id" is missing or invalid', () => {
        const types = [
          R.Recommendations.Type.AlsoViewed,
          R.Recommendations.Type.AlsoBought,
        ];
        const allowedItemIds = ['foo', '812', 456];
        types.forEach(type =>
          testValidatePresence(
            V.validateRecommendations,
            'item_id',
            allowedItemIds,
            { type }
          )
        );
      });
    });

    describe('frequently purchased together', () => {
      it('throws error if "item_ids" is missing or empty', () => {
        const type = R.Recommendations.Type.FrequentlyPurchasedTogether;
        const allowedItemIds = [['bar'], [123, 'baz-qux'], [456, 111]];
        testValidatePresence(
          V.validateRecommendations,
          'item_ids',
          allowedItemIds,
          { type }
        );
      });
    });
  });

  describe('validateFeedback', () => {
    describe('click suggestion, redirect', () => {
      it('throws error if required params is missing or invalid', () => {
        const allowed = ['bar', 32];
        const events = [
          R.Feedback.Event.ClickSuggestion,
          R.Feedback.Event.Redirect,
        ];
        events.forEach(event => {
          testValidatePresence(V.validateFeedback, 'rid', allowed, {
            event,
            suggestion: 'ok',
          });
          testValidatePresence(V.validateFeedback, 'suggestion', allowed, {
            event,
            rid: 'ok',
          });
        });
      });
    });

    describe('click item, add to cart', () => {
      it('throws error if required params is missing or invalid', () => {
        const allowed = ['bar', 32];
        const events = [R.Feedback.Event.ClickItem, R.Feedback.Event.AddToCart];
        events.forEach(event => {
          testValidatePresence(V.validateFeedback, 'item_id', allowed, {
            event,
          });
        });
      });
    });

    describe('purchase', () => {
      const event = R.Feedback.Event.Purchase;
      const allowed = ['bar', 32];
      const line_items = [];
      const requests = [
        {
          key: 'revenue',
          data: { order_id: 123, currency: 'usd', line_items },
        },
        {
          key: 'currency',
          data: { order_id: 123, revenue: 432, line_items },
        },
        {
          key: 'order_id',
          data: { revenue: 432, currency: 'eur', line_items },
        },
      ];
      requests.forEach(({ key, data }) => {
        const params = { ...data, event };
        testValidatePresence(V.validateFeedback, key, allowed, params);
      });
    });

    describe('purchase, update cart', () => {
      const events = [R.Feedback.Event.Purchase, R.Feedback.Event.UpdateCart];
      events.forEach(event => {
        const params = { event, order_id: 123, currency: 'usd', revenue: 5123 };
        testValidateLineItems(V.validateFeedback, params);
      });
    });

    describe('view page', () => {
      const event = R.Feedback.Event.ViewPage;
      const allowed = ['anything', 413];
      const requests = [
        {
          key: 'height',
          data: {
            url: 'https://example.com',
            ref: 'http://localhost',
            width: 320,
          },
        },
        {
          key: 'width',
          data: {
            url: 'http://example.com',
            ref: 'https://localhost:3001',
            height: 320,
          },
        },
        {
          key: 'ref',
          data: {
            url: 'http://example.com:8081',
            width: '313',
            height: 320,
          },
        },
        {
          key: 'url',
          data: {
            ref: 'https://localhost/foo/bar',
            width: 320,
            height: '10',
          },
        },
      ];
      requests.forEach(({ key, data }) => {
        const params = { ...data, event };
        testValidatePresence(V.validateFeedback, key, allowed, params);
      });
    });
  });

  function testValidateLineItems(fn: (x: any) => void, params: object = {}) {
    const allowed = ['bar', 32];
    const prohibited = [undefined, null, '', '   ', []];
    const withoutQuantity = prohibited.map(quantity => [
      { item_id: '123', unit_price: 71, quantity },
    ]);
    const withoutUnitPrice = prohibited.map(unit_price => [
      { item_id: 953, unit_price, quantity: 32 },
    ]);
    const withoutItemId = prohibited.map(item_id => [
      { item_id, unit_price: 38, quantity: '32' },
    ]);
    const invalid = [
      [{}, { item_id: '23', unit_price: 821 }],
      [{ item_id: '123' }],
      ...withoutQuantity,
      ...withoutUnitPrice,
      ...withoutItemId,
    ];
    const withQuantity = allowed.map(quantity => [
      { item_id: '123', unit_price: 71, quantity },
    ]);
    const withUnitPrice = allowed.map(unit_price => [
      { item_id: 953, unit_price, quantity: 32 },
    ]);
    const withItemId = allowed.map(item_id => [
      { item_id, unit_price: 38, quantity: '32' },
    ]);
    const valid = [...withQuantity, ...withUnitPrice, ...withItemId];
    invalid.forEach(line_items =>
      expect(() => fn({ ...params, line_items })).toThrowErrorMatchingSnapshot()
    );
    valid.forEach(line_items =>
      expect(() => fn({ ...params, line_items })).not.toThrow()
    );
  }

  function testValidatePresence(
    fn: (x: any) => void,
    key: string,
    allowed: any[],
    params: object = {}
  ) {
    const prohibited = [undefined, null, '', '   ', []];
    const invalid = prohibited.map(v => ({ ...params, [key]: v }));
    const valid = allowed.map(v => ({ ...params, [key]: v }));
    invalid.forEach(v => expect(() => fn(v)).toThrowErrorMatchingSnapshot());
    valid.forEach(v => expect(() => fn(v)).not.toThrow());
  }

  function testValidateList(fn: (x: any) => void, params: object = {}) {
    const valid = [
      {},
      {
        filters: [{ name: 'foo', type: 'whatever' }],
        offset: 40,
      },
      {
        filters: [
          { type: 'qux', name: 'smth', values: [{ from: '100', to: '500' }] },
          { type: 'quux', name: 'anything' },
        ],
        sort: [
          { field: 'foo', order: SortingOrder.Asc },
          { field: 'bar', order: SortingOrder.Desc },
        ],
        offset: 10,
        limit: 20,
      },
      {
        sort: [
          { field: 'foo', order: SortingOrder.Asc },
          { field: 'bar', order: SortingOrder.Desc },
        ],
        limit: 10,
      },
    ];
    const invalid = [
      { filters: [{ name: 'foo' }] },
      { filters: [{ type: 'bar' }] },
      {
        filters: [
          { type: 'qux', values: [{ from: '100', to: '500' }] },
          { type: 'quux' },
        ],
        limit: 10,
      },
      { sort: [{ field: 'grault' }], offset: 20 },
      { sort: [{ order: SortingOrder.Asc }], limit: 40, offset: 30 },
      { sort: [{ field: 'corge', order: 'bad' }] },
      {
        filters: [
          { type: 'quux' },
          { name: 'bar', values: [{ value: '100500' }] },
        ],
      },
    ];
    invalid.forEach(req =>
      expect(() => fn({ ...params, ...req })).toThrowErrorMatchingSnapshot()
    );
    valid.forEach(req => expect(() => fn({ ...params, ...req })).not.toThrow());
  }
});
