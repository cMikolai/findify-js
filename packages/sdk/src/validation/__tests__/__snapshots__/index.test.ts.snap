// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"\\"revenue\\" param is required"`;

exports[` 2`] = `"\\"revenue\\" param is required"`;

exports[` 3`] = `"\\"revenue\\" param can not be an empty or only-whitespace string"`;

exports[` 4`] = `"\\"revenue\\" param can not be an empty or only-whitespace string"`;

exports[` 5`] = `"\\"revenue\\" param can not be empty"`;

exports[` 6`] = `"\\"currency\\" param is required"`;

exports[` 7`] = `"\\"currency\\" param is required"`;

exports[` 8`] = `"\\"currency\\" param can not be an empty or only-whitespace string"`;

exports[` 9`] = `"\\"currency\\" param can not be an empty or only-whitespace string"`;

exports[` 10`] = `"\\"currency\\" param can not be empty"`;

exports[` 11`] = `"\\"order_id\\" param is required"`;

exports[` 12`] = `"\\"order_id\\" param is required"`;

exports[` 13`] = `"\\"order_id\\" param can not be an empty or only-whitespace string"`;

exports[` 14`] = `"\\"order_id\\" param can not be an empty or only-whitespace string"`;

exports[` 15`] = `"\\"order_id\\" param can not be empty"`;

exports[` 16`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 17`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 18`] = `"\\"line_items[].quantity\\" param is required"`;

exports[` 19`] = `"\\"line_items[].quantity\\" param is required"`;

exports[` 20`] = `"\\"line_items[].quantity\\" param can not be an empty or only-whitespace string"`;

exports[` 21`] = `"\\"line_items[].quantity\\" param can not be an empty or only-whitespace string"`;

exports[` 22`] = `"\\"line_items[].quantity\\" param can not be empty"`;

exports[` 23`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 24`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 25`] = `"\\"line_items[].unit_price\\" param can not be an empty or only-whitespace string"`;

exports[` 26`] = `"\\"line_items[].unit_price\\" param can not be an empty or only-whitespace string"`;

exports[` 27`] = `"\\"line_items[].unit_price\\" param can not be empty"`;

exports[` 28`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 29`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 30`] = `"\\"line_items[].item_id\\" param can not be an empty or only-whitespace string"`;

exports[` 31`] = `"\\"line_items[].item_id\\" param can not be an empty or only-whitespace string"`;

exports[` 32`] = `"\\"line_items[].item_id\\" param can not be empty"`;

exports[` 33`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 34`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 35`] = `"\\"line_items[].quantity\\" param is required"`;

exports[` 36`] = `"\\"line_items[].quantity\\" param is required"`;

exports[` 37`] = `"\\"line_items[].quantity\\" param can not be an empty or only-whitespace string"`;

exports[` 38`] = `"\\"line_items[].quantity\\" param can not be an empty or only-whitespace string"`;

exports[` 39`] = `"\\"line_items[].quantity\\" param can not be empty"`;

exports[` 40`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 41`] = `"\\"line_items[].unit_price\\" param is required"`;

exports[` 42`] = `"\\"line_items[].unit_price\\" param can not be an empty or only-whitespace string"`;

exports[` 43`] = `"\\"line_items[].unit_price\\" param can not be an empty or only-whitespace string"`;

exports[` 44`] = `"\\"line_items[].unit_price\\" param can not be empty"`;

exports[` 45`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 46`] = `"\\"line_items[].item_id\\" param is required"`;

exports[` 47`] = `"\\"line_items[].item_id\\" param can not be an empty or only-whitespace string"`;

exports[` 48`] = `"\\"line_items[].item_id\\" param can not be an empty or only-whitespace string"`;

exports[` 49`] = `"\\"line_items[].item_id\\" param can not be empty"`;

exports[` 50`] = `"\\"height\\" param is required"`;

exports[` 51`] = `"\\"height\\" param is required"`;

exports[` 52`] = `"\\"height\\" param can not be an empty or only-whitespace string"`;

exports[` 53`] = `"\\"height\\" param can not be an empty or only-whitespace string"`;

exports[` 54`] = `"\\"height\\" param can not be empty"`;

exports[` 55`] = `"\\"width\\" param is required"`;

exports[` 56`] = `"\\"width\\" param is required"`;

exports[` 57`] = `"\\"width\\" param can not be an empty or only-whitespace string"`;

exports[` 58`] = `"\\"width\\" param can not be an empty or only-whitespace string"`;

exports[` 59`] = `"\\"width\\" param can not be empty"`;

exports[` 60`] = `"\\"ref\\" param is required"`;

exports[` 61`] = `"\\"ref\\" param is required"`;

exports[` 62`] = `"\\"ref\\" param can not be an empty or only-whitespace string"`;

exports[` 63`] = `"\\"ref\\" param can not be an empty or only-whitespace string"`;

exports[` 64`] = `"\\"ref\\" param can not be empty"`;

exports[` 65`] = `"\\"url\\" param is required"`;

exports[` 66`] = `"\\"url\\" param is required"`;

exports[` 67`] = `"\\"url\\" param can not be an empty or only-whitespace string"`;

exports[` 68`] = `"\\"url\\" param can not be an empty or only-whitespace string"`;

exports[` 69`] = `"\\"url\\" param can not be empty"`;

exports[`validation validateAutocomlete throws error if "q" is missing or invalid 1`] = `"\\"q\\" param is required"`;

exports[`validation validateAutocomlete throws error if "q" is missing or invalid 2`] = `"\\"q\\" param is required"`;

exports[`validation validateAutocomlete throws error if "q" is missing or invalid 3`] = `"\\"q\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateAutocomlete throws error if "q" is missing or invalid 4`] = `"\\"q\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateAutocomlete throws error if "q" is missing or invalid 5`] = `"\\"q\\" param can not be empty"`;

exports[`validation validateConfig throws an error if "key" is missing or invalid 1`] = `"\\"key\\" param is missing. To get a Merchant API key you need to register with Findify and finish the onboarding process."`;

exports[`validation validateConfig throws an error if "key" is missing or invalid 2`] = `"\\"key\\" param should be a string. To get a Merchant API key you need to register with Findify and finish the onboarding process."`;

exports[`validation validateConfig throws an error if "key" is missing or invalid 3`] = `"\\"key\\" param should be a string. To get a Merchant API key you need to register with Findify and finish the onboarding process."`;

exports[`validation validateConfig throws an error if "key" is missing or invalid 4`] = `"\\"key\\" param can not be an empty or only-whitespace string. To get a Merchant API key you need to register with Findify and finish the onboarding process."`;

exports[`validation validateConfig throws an error if "key" is missing or invalid 5`] = `"\\"key\\" param can not be an empty or only-whitespace string. To get a Merchant API key you need to register with Findify and finish the onboarding process."`;

exports[`validation validateConfig throws an error if "log" is invalid 1`] = `"\\"log\\" param should be \\"boolean\\" type"`;

exports[`validation validateConfig throws an error if "log" is invalid 2`] = `"\\"log\\" param should be \\"boolean\\" type"`;

exports[`validation validateConfig throws an error if "log" is invalid 3`] = `"\\"log\\" param should be \\"boolean\\" type"`;

exports[`validation validateConfig throws an error if "method" is invalid 1`] = `"\\"method\\" param should be either \\"post\\" or \\"jsonp\\""`;

exports[`validation validateConfig throws and error if trying to use jsonp when not in browser 1`] = `"jsonp method is not allowed in node environment"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 1`] = `"\\"item_id\\" param is required"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 2`] = `"\\"item_id\\" param is required"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 3`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 4`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 5`] = `"\\"item_id\\" param can not be empty"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 6`] = `"\\"item_id\\" param is required"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 7`] = `"\\"item_id\\" param is required"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 8`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 9`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click item, add to cart throws error if required params is missing or invalid 10`] = `"\\"item_id\\" param can not be empty"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 1`] = `"\\"rid\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 2`] = `"\\"rid\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 3`] = `"\\"rid\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 4`] = `"\\"rid\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 5`] = `"\\"rid\\" param can not be empty"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 6`] = `"\\"suggestion\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 7`] = `"\\"suggestion\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 8`] = `"\\"suggestion\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 9`] = `"\\"suggestion\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 10`] = `"\\"suggestion\\" param can not be empty"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 11`] = `"\\"rid\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 12`] = `"\\"rid\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 13`] = `"\\"rid\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 14`] = `"\\"rid\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 15`] = `"\\"rid\\" param can not be empty"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 16`] = `"\\"suggestion\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 17`] = `"\\"suggestion\\" param is required"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 18`] = `"\\"suggestion\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 19`] = `"\\"suggestion\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateFeedback click suggestion, redirect throws error if required params is missing or invalid 20`] = `"\\"suggestion\\" param can not be empty"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 1`] = `"\\"item_id\\" param is required"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 2`] = `"\\"item_id\\" param is required"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 3`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 4`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 5`] = `"\\"item_id\\" param can not be empty"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 6`] = `"\\"item_id\\" param is required"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 7`] = `"\\"item_id\\" param is required"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 8`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 9`] = `"\\"item_id\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations also viewed / bought throws error if "item_id" is missing or invalid 10`] = `"\\"item_id\\" param can not be empty"`;

exports[`validation validateRecommendations frequently purchased together throws error if "item_ids" is missing or empty 1`] = `"\\"item_ids\\" param is required"`;

exports[`validation validateRecommendations frequently purchased together throws error if "item_ids" is missing or empty 2`] = `"\\"item_ids\\" param is required"`;

exports[`validation validateRecommendations frequently purchased together throws error if "item_ids" is missing or empty 3`] = `"\\"item_ids\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations frequently purchased together throws error if "item_ids" is missing or empty 4`] = `"\\"item_ids\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations frequently purchased together throws error if "item_ids" is missing or empty 5`] = `"\\"item_ids\\" param can not be empty"`;

exports[`validation validateRecommendations slot throws error if "slot" is missing or invalid 1`] = `"\\"slot\\" param is required"`;

exports[`validation validateRecommendations slot throws error if "slot" is missing or invalid 2`] = `"\\"slot\\" param is required"`;

exports[`validation validateRecommendations slot throws error if "slot" is missing or invalid 3`] = `"\\"slot\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations slot throws error if "slot" is missing or invalid 4`] = `"\\"slot\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations slot throws error if "slot" is missing or invalid 5`] = `"\\"slot\\" param can not be empty"`;

exports[`validation validateRecommendations throws error if recommendation type is invalid 1`] = `"Invalid \\"recommendation\\" type"`;

exports[`validation validateRecommendations throws error if recommendations type is missing 1`] = `"\\"type\\" param is required"`;

exports[`validation validateRecommendations throws error if recommendations type is missing 2`] = `"\\"type\\" param is required"`;

exports[`validation validateRecommendations throws error if recommendations type is missing 3`] = `"\\"type\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateRecommendations throws error if recommendations type is missing 4`] = `"\\"type\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateSearch throws error if request parameters is invalid 1`] = `"\\"q\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 2`] = `"\\"q\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 3`] = `"\\"q\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateSearch throws error if request parameters is invalid 4`] = `"\\"q\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateSearch throws error if request parameters is invalid 5`] = `"\\"q\\" param can not be empty"`;

exports[`validation validateSearch throws error if request parameters is invalid 6`] = `"\\"filter.type\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 7`] = `"\\"filter.name\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 8`] = `"\\"filter.name\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 9`] = `"\\"sort.order\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 10`] = `"\\"sort.field\\" param is required"`;

exports[`validation validateSearch throws error if request parameters is invalid 11`] = `"\\"sort.order\\" param should be either \\"asc\\" or \\"desc\\""`;

exports[`validation validateSearch throws error if request parameters is invalid 12`] = `"\\"filter.name\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 1`] = `"\\"slot\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 2`] = `"\\"slot\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 3`] = `"\\"slot\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 4`] = `"\\"slot\\" param can not be an empty or only-whitespace string"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 5`] = `"\\"slot\\" param can not be empty"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 6`] = `"\\"filter.type\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 7`] = `"\\"filter.name\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 8`] = `"\\"filter.name\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 9`] = `"\\"sort.order\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 10`] = `"\\"sort.field\\" param is required"`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 11`] = `"\\"sort.order\\" param should be either \\"asc\\" or \\"desc\\""`;

exports[`validation validateSmartCollection throws error if request parameters is invalid 12`] = `"\\"filter.name\\" param is required"`;
